File: LinkedIn/Core/LinkedIn.Domain/Entities/LinkedInPost.cs
using SNM.LinkedIn.Domain.Common;
using SNM.LinkedIn.Domain.Entities;
using SNM.LinkedIn.Domain.Enumeration;

public class LinkedInPost : EntityBase<Guid>
{

    public Guid? PostId { get; set; }
 
    //public string Photo { get; set; }
    //public string Video { get; set; }
    //public string MediaUrn { get; set; }
    //public string MediaUrl { get; set; }
    public string PostUrn { get; set; }

    public PublicationStatusEnum PublicationStatus { get; set; }
    public DateTime? PublicationDate { get; set; }
    public string Message { get; set; }

   // public LinkedInInsight insight { get; set; }
    public Guid LinkedInChannelId { get; set; }
    public virtual LinkedInChannel LinkedInChannel { get; set; }
    public virtual Post? Post { get; set; }


}

File: SNM.FrontEnd/src/app/core/models/linkedin-post.ts
import { LinkedInInsight } from "./linkedin-insight";

export interface LinkedinPost {

    Id: string
    postId: string;
    author_urn: string; //person or organization
    author_username: string
    author_profilelink: string
    photo: string
    video: string
    message: string;
    createdAt: Date;
    mediaUrn: string;
    mediaUrl: string;
    insight: LinkedInInsight;




}
